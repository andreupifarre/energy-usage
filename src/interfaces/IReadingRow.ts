export interface IReadingRow {
  readingDate: string;
  cumulative: number;
  unit: string;
}
