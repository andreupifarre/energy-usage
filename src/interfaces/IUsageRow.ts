export interface IUsageRow {
  date: string;
  energyUsage: number;
}
