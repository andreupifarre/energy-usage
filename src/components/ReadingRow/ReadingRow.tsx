import * as React from 'react';
import { IReadingRow } from '../../interfaces/IReadingRow';

const ReadingRow: React.SFC<IReadingRow> = ({
  readingDate,
  cumulative,
  unit
}) => (
  <tr>
    <td>{readingDate}</td>
    <td>{cumulative}</td>
    <td>{unit}</td>
  </tr>
);

export default ReadingRow;
