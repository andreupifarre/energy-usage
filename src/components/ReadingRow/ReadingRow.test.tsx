import * as React from 'react';
import { shallow } from 'enzyme';
import ReadingRow from './ReadingRow';
import { IReadingRow } from '../../interfaces/IReadingRow';

const render = props => shallow(<ReadingRow {...props} />);

describe('ReadingRow', () => {
  it('should match the props values passed with the value rendered for each column', () => {
    const reading: IReadingRow = {
      readingDate: '2017-03-28T00:00:00.000Z',
      cumulative: 17580,
      unit: 'kWh'
    };

    const cols = render(reading).find('td');
    const newCols = cols.map(col => col.text());
    const newReading = Object.keys(reading).map(key => `${reading[key]}`);

    expect(newCols).toEqual(newReading);
  });

  it('should render correctly', () => {
    const reading: IReadingRow = {
      readingDate: '2017-03-28T00:00:00.000Z',
      cumulative: 17580,
      unit: 'kWh'
    };

    expect(render(reading).getElements()).toMatchSnapshot();
  });
});
