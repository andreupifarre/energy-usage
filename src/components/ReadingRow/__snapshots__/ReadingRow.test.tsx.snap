// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingRow should render correctly 1`] = `
Array [
  <tr>
    <td>
      2017-03-28T00:00:00.000Z
    </td>
    <td>
      17580
    </td>
    <td>
      kWh
    </td>
  </tr>,
]
`;
